### PR TITLE
TransactionBatch DI support quality of life improvements

### PR DESCRIPTION
**Pull request by boblangley**

### CODE REVIEW

File: src/SharedAcceptanceTests.RequirePartitionKey/When_using_outbox_synchronized_session_via_container.cs [hunk 1]
```diff
@@ -34,14 +34,10 @@ public Endpoint()
                 EndpointSetup<DefaultServer>(config =>
                 {
                     config.EnableOutbox();
+                    config.UsePersistence<CosmosDbPersistence>().RegisterSharedTransactionalBatchForDependencyInjection();
```
Comment (SeanFeldman) [inline src/SharedAcceptanceTests.RequirePartitionKey/When_using_outbox_synchronized_session_via_container.cs:37]: @danielmarbach, there was a way to access the currently used persistence w/o calling `UsePersistence<>` again. What was it?

**Review by SeanFeldman — APPROVED** [review]

**Review by danielmarbach — COMMENTED** [review]
Why was this done? There was a reason I introduced this interface.

A) compliant with other persisted approaches
b) registering an SDK type directly is not a good idea because customers can already have it registered
c) in order to address the above you had to introduce another setting method

I think changes like that should be discussed first before merged. I believe this should be rolled back

**Comment by danielmarbach** [comment]
Raised https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/41

**Comment by boblangley** [comment]
As discussed we will rework #41 only to not revert the extension method change introduced here, but revert all of the other changes.

**Pull request merged**